### PR TITLE
added utils file for json response and password hashing validation

### PR DESCRIPTION
**Pull request by Armolas**
**Additions:**

- I added a `json_response` file in the utils directory for easy formatting of the JSON response
- I added a `password_auth` file in the utils directory that contains function for hashing and validating passwords
- I added a `BaseModel` class to all the objects which contains the following methods:

> - `to_dict()`: returns a dictionary representation of the object
> - `get_all()`: returns a list of all the object of the class present in the database
> - `get_by_id(id: str)`: returns a single object from the database whose id is passed as argument

### CODE REVIEW

File: api/utils/json_response.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+from enum import Enum
+from json import dumps
+from fastapi import status
+from fastapi.responses import JSONResponse
+from fastapi.encoders import jsonable_encoder
+
+
+class RequestStatus(Enum):
+    SUCCESS = "success"
+    ERROR = "error"
```
Comment (Dev-wonderful) [inline api/utils/json_response.py:13]: Not necessary for the accepted convention

File: api/utils/json_response.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        })
+
+    def response(self):
+        """return a json response dictionary"""
+        print(f"response: {format(self)}")
+        if self.status == "success":
+            return {
+                "message": self.message,
+                "data": self.data,
+                "status": self.status
```
Comment (Dev-wonderful) [inline api/utils/json_response.py:50]: Based on the enforced convention, it should be `message`, `data` and `status_code`

File: api/v1/models/base.py [hunk 3]
```diff
@@ -19,3 +19,33 @@
...
+            obj_dict["created_at"] = self.created_at.isoformat()
+        if self.updated_at:
+            obj_dict["updated_at"] = self.updated_at.isoformat()
+        return obj_dict
+
+    @classmethod
+    def get_all(cls):
+        """ returns all instance of the class in the db
+        """
+        from api.db.database import db
```
Comment (Dev-wonderful) [inline api/v1/models/base.py:42]: Why?

**Review by Dev-wonderful — CHANGES_REQUESTED** [review]

File: api/v1/models/base.py (hunk 3 above)
Comment (Armolas) [reply api/v1/models/base.py:42]: To prevent circular import error boss

File: .env.sample [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
-DB_PORT=5432
-MYSQL_DRIVER=pymysql 
-DB_URL="postgresql://user:password@127.0.0.1:5432/dbname"
+DB_NAME=postgres
+DB_USER=postgres.papbnyufdxepjdnumjdx
+DB_PASSWORD=be_fast_api
+DB_HOST=aws-0-us-east-1.pooler.supabase.com
+DB_PORT=6543
+MYSQL_DRIVER=
+DB_URL=postgresql://postgres.papbnyufdxepjdnumjdx:be_fast_api@aws-0-us-east-1.pooler.supabase.com:6543/postgres
```
[the remark this answers is not in the document]
Comment (Armolas) [reply .env.sample:9]: Okay boss

File: api/utils/json_response.py (hunk 1 above)
Comment (Armolas) [reply api/utils/json_response.py:13]: Should I remove it?

Comment (Dev-wonderful) [reply api/utils/json_response.py:13]: Yes

File: api/v1/models/base.py (hunk 3 above)
Comment (Dev-wonderful) [reply api/v1/models/base.py:42]: What is causing it this early in the project

**Pull request closed without merging**